backend/nokia-innovative-project/src/test/java/com/nokia/library/nokiainnovativeproject/repositories/RepositoriesTests.java

```diff
@@ -0,0 +1,65 @@
+package com.nokia.library.nokiainnovativeproject.repositories;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.nokia.library.nokiainnovativeproject.entities.Book;
+import com.nokia.library.nokiainnovativeproject.repositories.BookRepository;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class RepositoriesTests {
+	@Autowired
+    private BookRepository bookRepository;
+	
+	@Test
+	public void whenAddBookThenBookAdded() {

```

Comment:
There is no sense to load whole application context to test only repositories. 
Context should be used for e2e test or integration test. 

There is also no big sense to test repositories.  It is part of the code that was provided by Spring. I think that we can trust Spring developers. ;) 

You can write a test that will check your controller and database in one test. 
For example: 
test checks wheather after sending POST to endpoint `/api/v1/books` with body:
```
{
"title" : "book"
}
```
an appropriate entry in the database will be created after making the test